### ISSUE TITLE
Multiple instantsearch instances and URL Syncing

### ISSUE THREAD

**Issue by lukasavic** [algolia/instantsearch.js#1660]
I have a question regarding the multiple instantsearch instances on the same page and URL syncing. 

We use one instance for "global" search and we also have "category" search that will search indices only by the current category. 
The problem is that URL syncing is working only for one of the search instances, but we want this to work for both of them. 

Is something like that possible?

**Comment by mthuret** [comment]
Hi @lukasavic,

Which version of react-instantsearch are you currently using? 

On the last version, the URL Syncing has to be implemented by the user typically to allow more advance use cases like yours. To synchronize an `InstantSearch` with an URL you need to handle the `InstantSearch` state externally like described here: https://github.com/algolia/instantsearch.js/blob/v2/docgen/src/guide/Routing.md. 
You can also find the guide about multiple indices here: https://github.com/algolia/instantsearch.js/blob/v2/docgen/src/guide/Multi%20index.md

Be careful, the API described in those guides has not been entirely released yet so I suggest you wait for a couple of days before trying it. Those guides will also be available [on our documentation website](https://community.algolia.com/instantsearch.js/react/).

Basically the idea is to listen to every state changes for both indices and create your URL accordingly. Then, when you'll need to update the `InstantSearch` state, you will take your url and  reconstruct the state based on your previous conventions.

**Comment by lukasavic** [comment]
Hey @mthuret 

tnx for the feedback.
Currently I'm  using instantsearch.js v 1.8.16 ( not react-instantsearch ) sorry if I wasn't clear at the begining. So the URL Syncing works, but only for one instance on the page. The idea is to make them both work. I was thinking of using "searchFunction" in "urlSync" property and then manually update the history state, but I would need to create a valid url from the helper.state in searchFunction. I there any way to do this and if you can point me to the right soulution for this?

**Comment by mthuret** [comment]
I'm sorry for the misunderstanding @lukasavic. Unfortunately there's no easy way to achieve what you want with instantsearch.js as the urlSync feature was design with only one instance in mind. We're going to work on it in the future though. Thanks for reporting it.

**Comment by lukasavic** [comment]
@mthuret tnx for the info. I'm closing this issue.

**Comment by sheikhkitabb** [comment]
has this url synch been solves now after 9 years , i have installed lates v7 now but i have two tabs with seperate instantsearch components separately in them but when i switch back and forth using navigation in nextjs paginatino stops applying to url but in ui it is working

**Comment by sheikhkitabb** [comment]
Thanks will try hoisting it
